config/initializers/omniauth.rb

```diff
@@ -35,13 +12,47 @@
              path_prefix: "/check-records/auth"
   end
 else
-  Rails.application.config.middleware.use OmniAuth::Strategies::OpenIDConnect, options
+  dfe_sign_in_issuer_uri =
+    ENV["DFE_SIGN_IN_ISSUER"].present? ? URI(ENV["DFE_SIGN_IN_ISSUER"]) : nil
+
+  Rails.application.config.middleware.use OmniAuth::Builder do
+    provider :openid_connect,
+             name: :dfe,
+             callback_path: "/check-records/auth/dfe/callback",
+             client_options: {
+               host: dfe_sign_in_issuer_uri&.host,
+               identifier: ENV.fetch("DFE_SIGN_IN_CLIENT_ID", "example"),
+               port: dfe_sign_in_issuer_uri&.port,
+               redirect_uri: ENV.fetch("DFE_SIGN_IN_REDIRECT_URL", "example"),
+               scheme: dfe_sign_in_issuer_uri&.scheme,
+               secret: ENV.fetch("DFE_SIGN_IN_SECRET", "example")
+             },
+             discovery: true,
+             issuer: ENV.fetch("DFE_SIGN_IN_ISSUER", "example"),
```

Comment:
This has quite a specific expected format — need the port number appended like in the original code

Reply:
Yep, Identity and DSI have different configurations, can view the issuer value for both at (for example):

https://dev-oidc.signin.education.gov.uk/.well-known/openid-configuration
https://preprod.teaching-identity.education.gov.uk/.well-known/openid-configuration